pkg/cmd/util/clientcmd/factory.go

```diff
@@ -569,6 +539,59 @@ func (w *Factory) ApproximatePodTemplateForObject(object runtime.Object) (*api.P
 	}
 }
 
+// PodForRemoteShell resolves the passed resource and returns
+func (f *Factory) PodForRemoteShell(resource string) (string, error) {
+	namespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return "", err
+	}
+	oc, kc, err := f.Clients()
+	if err != nil {
+		return "", err
+	}
+	mapper, _ := f.Object()
+	resourceType, name, err := util.ResolveResource("pods", resource, mapper)
+	if err != nil {
+		return "", err
+	}
+
+	switch resourceType {
+	case "pods":
+	case "replicationcontrollers":
+		rc, err := kc.ReplicationControllers(namespace).Get(name)
+		if err != nil {
+			return "", err
+		}
+		if rc.Spec.Replicas == 0 {
+			return "", fmt.Errorf("cannot get remote shell to %s; no running replicas found", rc.Name)
+		}
+		selector := labels.SelectorFromSet(rc.Spec.Selector)
+		pod, _, err := cmdutil.GetFirstPod(kc, namespace, selector)
+		if err != nil {
+			return "", err
+		}
+		return pod.Name, nil
+	case "deploymentconfigs":
+		dc, err := oc.DeploymentConfigs(namespace).Get(name)
+		if err != nil {
+			return "", err
+		}
+		if dc.Spec.Replicas == 0 {
+			return "", fmt.Errorf("cannot get remote shell to %s; no running replicas found", dc.Name)
+		}
+		selector := labels.SelectorFromSet(dc.Spec.Selector)
+		pod, _, err := cmdutil.GetFirstPod(kc, namespace, selector)
+		if err != nil {
+			return "", err
+		}
+		return pod.Name, nil
+	default:
```

Comment:
actually, you won't be able to do it as assemble script is managed by s2i... nvmd.